packages/core/src/util/path.ts

```diff
@@ -52,10 +52,14 @@ export { compose as composePaths };
  * and de-referencing the single segments to obtain a new object.
  *
  *
- * @param {string} schemaPath the schema path to be converted
+ * @param {string?} schemaPath the schema path to be converted
  * @returns {string[]} an array containing only non-schema-specific segments
  */
-export const toDataPathSegments = (schemaPath: string): string[] => {
+export const toDataPathSegments = (schemaPath?: string): string[] => {
+  if (schemaPath === null) {
+    return [];
+  }
```

Comment:
You can see that the typings don't match. The parameter is typed ad `string | undefined` but now we also check for `null`. This should be avoided.

Reply:
You are right